File: quickstart/python/hello.py
"""Cryptohopper Python SDK — hello world.

Authenticates with a bearer token from $CRYPTOHOPPER_TOKEN, then makes
two calls: user.get() and exchange.ticker().

Run:
    export CRYPTOHOPPER_TOKEN=your-40-char-bearer
    pip install -r requirements.txt
    python hello.py
"""

import os
import sys

from cryptohopper import CryptohopperClient, CryptohopperError


def main() -> int:
    token = os.environ.get("CRYPTOHOPPER_TOKEN")
    if not token:
        print("Set CRYPTOHOPPER_TOKEN to a 40-char OAuth bearer first.", file=sys.stderr)
        return 1

    try:
        with CryptohopperClient(api_key=token) as ch:
            me = ch.user.get()
            print(f"Logged in as {me.get('email') or me.get('username') or me.get('id')}")

            ticker = ch.exchange.ticker(exchange="binance", market="BTC/USDT")
            print(f"BTC/USDT last: {ticker.get('last')}")
    except CryptohopperError as e:
        print(f"API error [{e.code}]: {e}", file=sys.stderr)
        if e.ip_address:
            print(f"  Server saw your IP as: {e.ip_address}", file=sys.stderr)
        return 2

    return 0


if __name__ == "__main__":
    sys.exit(main())

File: quickstart/dart/bin/hello.dart
// Cryptohopper Dart SDK — hello world.
//
// Authenticates with a bearer token from $CRYPTOHOPPER_TOKEN, then makes
// two calls: user.get() and exchange.ticker().
//
// Run:
//   export CRYPTOHOPPER_TOKEN=your-40-char-bearer
//   dart pub get
//   dart run

import 'dart:io';

import 'package:cryptohopper/cryptohopper.dart';

Future<void> main() async {
  final token = Platform.environment['CRYPTOHOPPER_TOKEN'];
  if (token == null || token.isEmpty) {
    stderr.writeln('Set CRYPTOHOPPER_TOKEN to a 40-char OAuth bearer first.');
    exitCode = 1;
    return;
  }

  final ch = CryptohopperClient(apiKey: token);
  try {
    final me = await ch.user.get() as Map<String, dynamic>;
    final identity = me['email'] ?? me['username'] ?? me['id'] ?? '?';
    print('Logged in as $identity');

    final ticker = await ch.exchange.ticker(
      exchange: 'binance',
      market: 'BTC/USDT',
    ) as Map<String, dynamic>;
    print('BTC/USDT last: ${ticker['last']}');
  } on CryptohopperException catch (e) {
    stderr.writeln('API error [${e.code}]: ${e.message}');
    if (e.ipAddress != null) {
      stderr.writeln('  Server saw your IP as: ${e.ipAddress}');
    }
    exitCode = 2;
  } finally {
    ch.close();
  }
}
